index.ts

```diff
@@ -105,15 +112,25 @@ export class SilentPayment {
    */
   static _ser32(i: number): Buffer {
     const returnValue = Buffer.allocUnsafe(4);
-    returnValue.writeUInt32LE(i);
-    return returnValue.reverse();
+    returnValue.writeUInt32BE(i);
+    return returnValue;
   }
 
   private static _sumPrivkeys(utxos: UTXO[]): Buffer {
     let ret = ECPair.fromWIF(utxos[0].WIF).privateKey;
 
+    // If taproot, check if the seckey results in an odd y-value and negate if so
+    if (utxos[0].is_taproot && sec.publicKeyCreate(ret)[0] === 0x03) {
+      ret = sec.privateKeyNegate(ret);
+    }
```

Comment:
is that a leftover from debug? checking only `utxos[0]` ?

Reply:
This is to check the first UTXO in the list, negate if necessary, and then iterate through the rest of the utxos (c = 1, ..) and add them up.

Reply:
why not iterate from 0 ?

Reply:
?

Reply:
ah sorry, missed this question. i guess we could re-write it to be more succinct, like a map/reduce. haha my javascript isn't great, so i just went with the simple version